test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -288,3 +306,133 @@ func TestVLANNetwork(t *testing.T) {
 	}
 	testSecondaryNetwork(t, networkTypeVLAN, pods)
 }
+
+func executeShellScript(script string) (string, error) {
+	cmd := exec.Command("bash", "-c", script)
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		return "", fmt.Errorf("failed to execute script: %s, output: %s, %s, error: %v", script, out.String(), stderr.String(), err)
+	}
+
+	return out.String(), nil
+}
+
+func (data *testData) assignIP() error {
+	e2eTestData := data.e2eTestData
+	namespace := e2eTestData.GetTestNamespace()
+
+	for _, testPod := range data.pods {
+		_, err := e2eTestData.PodWaitFor(defaultTimeout, testPod.podName, namespace, func(pod *corev1.Pod) (bool, error) {
+			if pod.Status.Phase != corev1.PodRunning {
+				return false, nil
+			}
+			podIPs, err := data.listPodIPs(testPod)
+			if err != nil {
+				return false, err
+			}
+			ip, exists := podIPs["eth1"]
+			if !exists || ip == nil {
+				logs.Infof("IP not available for interface 'eth1' in Pod %s, retrying...", testPod.podName)
+				return false, nil
+			}
+
+			cmd := fmt.Sprintf("aws ec2 assign-private-ip-addresses --network-interface-id %s --private-ip-addresses %s", testPod.secondaryENI, ip)
+			output, err := executeShellScript(cmd)
+			if err != nil {
+				return false, err
+			}
+			logs.Infof("assign private ip addresses: %s", output)
+			return true, nil
+		})
+		if err != nil {
+			return fmt.Errorf("error when waiting for secondary IPs for Pod %+v: %v", testPod, err)
+		}
+	}
+	return nil
+}
+
+// createNetworkAttachmentDefinition creates a NetworkAttachmentDefinition in the specified namespace.
+func createNetworkAttachmentDefinition(client networkclient.Interface, namespace, name, config string) error {
+	nad := &netattachv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: map[string]string{"k8s.v1.cni.cncf.io/resourceName": "intel.com/intel_sriov_netdevice"},
+		},
+		Spec: netattachv1.NetworkAttachmentDefinitionSpec{
+			Config: config,
+		},
+	}
+
+	_, err := client.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Create(context.TODO(), nad, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create NetworkAttachmentDefinition: %v", err)
+	}
+
+	fmt.Printf("NetworkAttachmentDefinition %s created successfully in namespace %s\n", name, namespace)
```

Comment:
the function should have testing.T as a parameter and use t.Log to get consistent log style.